### PR TITLE
Low hanging fruit for milestone 0.1

### LINKED ISSUE #203: implement equality check for data and RDMs
This would be a quality of life improvement

### LINKED ISSUE #231: calc functions should pass Dataset obs_descriptors to new RDMs as pattern_descriptors
In the transition from Dataset to RDMs, currently only the dataset-wide descriptors are kept, and sometimes one specific descriptor that was used for determining the slice.

```python
import numpy
from rsatoolbox.data import Dataset
from rsatoolbox.rdm.calc import calc_rdm_euclid

ds = Dataset(
    measurements=numpy.random.rand(3, 2),
    descriptors=dict(foo='bar'),
    obs_descriptors=dict(abc=['a', 'b', 'c']),
    channel_descriptors=dict(name=['x', 'y'])
)
rdms = calc_rdm_euclid(ds)
print(rdms)
```

This currently gives:

```shell
rsatoolbox.rdm.RDMs
1 RDM(s) over 3 conditions

### SNIP

pattern_descriptors: 
index = [0, 1, 2]
pattern = [0 1 2]
```

But should have `abc` as a `pattern_descriptor`

### LINKED ISSUE #232: Colormap unification
We talked last time about a unification fo the colormaps used in the toolbox. We certainly want to keep the option for users to choose their own colormaps of course, but by default we should have a reasonably consistent setting

### LINKED ISSUE #250: copy() method for RDMs
Standard library `copy.deepcopy` could be an alternative but it's not guaranteed that it plays nice with all our arrays of arrays etc.

One use case if for the methods that operate in place (e.g. `reorder`), if you want to keep the original.

### LINKED ISSUE #267: calc_rdm_euclid should be calc_rdm_euclidean
**Is your feature request related to a problem? Please describe.**
The other `calc` methods' names correspond to the method argument string 

**Describe the solution you'd like**
`calc_rdm_euclidean` instead of `calc_rdm_euclid`

### LINKED ISSUE #268: Improve Contributing Guide
- add a TLDR
- Type Hints see #177 
- NaN policy

### LINKED ISSUE #278: Missing documentation for Visualisation
The visualisation section of the documentation has some major gaps: https://rsatoolbox.readthedocs.io/en/latest/visualization.html

### LINKED ISSUE #279: Improve readthedocs versioning inline with new release management
- remove the tag v3.0.0.post20201106 as rtd considers it the latest for stable (as opposed to GitHub)
- enable stable builds for rtd
- display version on all documentation pages

### PR DESCRIPTION
**Pull request by JasperVanDenBosch**
Various (hopefully) simpler issues that I'll try to finish before we move towards releasing v0.1

### operations

- [x] resolves #203
- [x] resolves #231
- [x] resolves #250 
- [x] resolves #267
- [x] `_calc_pairwise_differences`
- [x] `calc_unbalanced` remove unused functions 
- [x] `calc_unbalanced` pass descriptors

### CODE REVIEW

**Comment by JasperVanDenBosch** [comment]
`test_two_rdms_nan` stochastic failure shows up again
```
=========================== short test summary info ===========================
[199](https://github.com/rsagroup/rsatoolbox/actions/runs/3462718000/jobs/5782028238#step:11:200)
FAILED tests/test_model.py::TestConsistency::test_two_rdms_nan - AssertionError: 0.0022591653136752017 not less than 0.001 : regression fit differs from optimization fit for corr_cov
[200](https://github.com/rsagroup/rsatoolbox/actions/runs/3462718000/jobs/5782028238#step:11:201)
============ 1 failed, 272 passed, 7 warnings in 97.27s (0:01:37) =============
```

**Comment by JasperVanDenBosch** [comment]
Ready for review if anyone wants to have a look.

Codefactor and Codacy at this point disagree over how we should indent function signatures; at some point we may want to specify our linting preferences in a way that both services understand (or just get rid of one of them).

**Review by HeikoSchuett — APPROVED** [review]
This looks good to me, thanks for implementing these!

File: src/rsatoolbox/rdm/calc_unbalanced.py [hunk 1]
```diff
@@ -199,193 +181,3 @@ def calc_one_similarity(data_i, data_j,
         method_idx, noise=noise,
         prior_lambda=prior_lambda, prior_weight=prior_weight,
         weighting=weight_idx)
-
-
-def calc_one_dissimilarity_cv(dataset, descriptor, i_des, j_des,
```
Comment (JasperVanDenBosch) [inline src/rsatoolbox/rdm/calc_unbalanced.py:204]: @HeikoSchuett could you check that these are the functions that you meant for us to remove?

Comment (JasperVanDenBosch) [reply src/rsatoolbox/rdm/calc_unbalanced.py:204]: I'll just go ahead with this for now and we can always restore them if I went too far

**Pull request merged**